src/Symfony/Component/Dotenv/Dotenv.php

```diff
@@ -453,12 +448,7 @@ private function resolveCommands(string $value, array $loadedVars): string
                 $process->inheritEnvironmentVariables();
             }
 
-            $env = [];
-            foreach ($this->values as $name => $value) {
-                if (isset($loadedVars[$name]) || (!isset($_ENV[$name]) && !(isset($_SERVER[$name]) && !str_starts_with($name, 'HTTP_')))) {
-                    $env[$name] = $value;
-                }
-            }
+            $env = array_filter($this->values, static fn($value, $name): bool => isset($loadedVars[$name]) || (!isset($_ENV[$name]) && !(isset($_SERVER[$name]) && !str_starts_with($name, 'HTTP_'))), ARRAY_FILTER_USE_BOTH);
```

Comment:
turning a foreach into an array_map creates needless temporary arrays, that's not an improvement.

Reply:
Hum, I could be wrong but we're using `array_filter` here, in the end, we obtain only the values that we need 🤔 

Maybe I'm missing something but is there any consideration regarding performances or developer experience to prevent using `array_filter`? 🤔

Reply:
We already have a temporary array here (`$env`), so I don't follow that argument. That being said, I neither see big value in this change. The `foreach` loop was perfectly fine. This change has nothing to do with modernization, we're just swapping one way to doing something for another.

Reply:
I agree, that's just another way of doing it, If I may ask it, why prefering using `foreach` over a native function like `array_filter`, is there any consideration regarding the developer experience (everyone know how to use `foreach`, maybe introducing `array_filter` can be harder to new developers?)?

_That's just a questions about understanding how the code is built_ 😅

Reply:
The `array_filter` has been removed 🙂